### PR TITLE
Add support for `{read,write}v` syscalls

### LINKED ISSUE #50: `Handler::syscall` `readv` implementation should perform at most one VMEXIT
Currently the implementation performs `iovcnt` VMEXITs with `read` on each `iovec`, that is terrible for performance and ideally this should be performed via `syscall::Readv` or at least single sally.
The main issue is constructing an iterator over `&mut [u8]` from an `iovec` passed to us, since we have to validate each `iov` individually and it is currently a fallible operation.

### LINKED ISSUE #60: [Feature]: `writev` support
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

`writev` support should be added as part of #34 

### Acceptance Criteria

_No response_

### Suggestions for a technical implementation

_No response_

### PR DESCRIPTION
**Pull request by rvolosatovs**
Part of #34 
Closes #50 
Closes #60 

Still TODO:
- [x] ~map "opaque" `readv` from `Handler::syscall` to `syscall::Readv` for now I don't see a simple way to do this, I suggest to open a separate issue to tackle this later as technical debt. #50~

### CODE REVIEW

File: src/v2/src/guest/syscall/readv.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                                }
+                            })
+                        }),
+                    )
+                }
+                Some(res)
+            }
+            err => Some(err),
+        }
+    }
```
Comment (npmccallum) [inline src/v2/src/guest/syscall/readv.rs:96]: This is too indent heavy. Perhaps this would be better expressed by `if` rather than `match`?

**Review by npmccallum — DISMISSED** [review]

**Comment by rvolosatovs** [comment]
Let's see if https://github.com/enarx/sallyport/issues/50#issuecomment-1007422521 works

File: src/v2/src/guest/syscall/readv.rs (hunk 1 above)
Comment (rvolosatovs) [reply src/v2/src/guest/syscall/readv.rs:99]: I'd rather keep the match to catch unhandled cases at compile-time, but I extracted the collection function, which should hopefully look better

File: src/v2/src/guest/handler.rs [hunk 2]
```diff
@@ -557,6 +570,19 @@ impl<'a, P: Platform> Execute for Handler<'a, P> {
...
+                debug_assert_eq!(
+                    core::ptr::slice_from_raw_parts_mut(0xfeed as *mut u8, 42),
+                    core::mem::transmute(libc::iovec {
+                        iov_base: 0xfeed as _,
+                        iov_len: 42
+                    }),
+                );
+                // NOTE: This is only valid under assumption that `&mut [u8]` and `libc::iovec` have the
+                // same memory layout.
+                let iovs = self.platform.validate_slice_mut::<&mut [u8]>(iov, iovcnt)?;
```
Comment (rvolosatovs) [inline src/v2/src/guest/handler.rs:583]: Can the `Platform` trait implementation validate the lifetime of the individual `iovec` elements here?
If not, we probably need a `validate_iov_mut` instead

Comment (rvolosatovs) [reply src/v2/src/guest/handler.rs:594]: I don't think this will work - I will add `validate_iov*` instead

Comment (rvolosatovs) [reply src/v2/src/guest/handler.rs]: Done, see explanation in https://github.com/enarx/sallyport/pull/49/commits/32a9fb261cac5af78a1fa71af023878dfe6e5b7f

**Comment by haraldh** [comment]
Please test behavior, when exceeding the sallyport block size.

Also test, if one of the readv/writev iovecs has zero length.

**Review by haraldh — APPROVED** [review]
